### DISCUSSION TITLE
[0.73, 0.74] Experimental New Debugger — Known Issues and Feedback

### DISCUSSION THREAD

**Discussion by huntie** [react-native-community/discussions-and-proposals#733] [Feedback]
> [!Important]
> 
> This discussion thread covers React Native 0.73-0.75, which shipped with an experimental debugger frontend, but not with our next gen debugger stack rewrite.
>
> If you are using React Native 0.76 (including RCs), you are now on [**React Native DevTools**](https://github.com/react-native-community/discussions-and-proposals/discussions/819). This experience provides a fundamentally different and more reliable experience than every debugger shipped in previous versions.
> 
> If you are on React Native 0.76, please use the correct feedback thread:
> - https://github.com/react-native-community/discussions-and-proposals/discussions/819

### Introduction

In React Native 0.73, we're shipping a Technical Preview of our experimental new debugger experience (opt-in via `npx react-native start --experimental-debugger`). 

This comes with:
- A preview of our new debugger frontend.
- A new `j` to debug shortcut in React Native CLI.
- Some improvements to Hermes (providing the debugger backend) in 0.73.
- (Foundational work that moves the debugger launch flow into the React Native repo, enabling us to iterate faster.)

![image](https://github.com/react-native-community/discussions-and-proposals/assets/2547783/dff07dea-ba55-4d64-a95a-f898d78aa61b)

The new debugger is **experimental** and does not represent the desired stable launch of this experience in future — which we will be holding to a higher quality bar than the existing debugging offerings.

### What we haven't improved, yet!

The new debugger **does not yet come with** some fundamental improvements we are making to our backend debugging stack. Among other things, these future improvements aim to:
- Improve debugger connection reliability.
- Enable breakpoint persistence (and generally fix the behaviour of breakpoints).
- Enable a correct debugger lifecycle across app reloads.
    - Specifically, introducing a React Native CDP server that maintains a debugging session for the native app instance, rather than for the currently executing JS bundle.
- Enable us to connect React DevTools to app instances over CDP, and embed React DevTools frontend in the debugger.
- Enable us to build new debugging capabilities on top of this stack in future.

### Discussion points

⬆️ So, as of 0.73, the new debugger shares some of the reliability issues and traps with the current debugging architecture(!). We want to use this thread to:
- Proactively highlight the biggest pain points we know of (and have plans to address).
- Take feedback from users on any issues and missing capabilities when using the new debugger frontend.

**Comment by huntie** [comment]
## [Known issue] Crash on paused debugger

If you reload the app while the debugger is paused on a breakpoint, the app will crash.

- When relaunching the app, the existing debugger window won't reconnect.
- Launching a new debugger instance will display a paused debugger state — however may be at a position unrelated to the original breakpoint/`debugger` statement.

<img width="339" alt="image" src="https://github.com/react-native-community/discussions-and-proposals/assets/2547783/0ce6dbd9-1f3e-4daa-b844-f96dca694448">

**Comment by huntie** [comment]
## [Known issue] Interactively setting a breakpoint doesn't work after a reload

Setting a breakpoint interactively (via the **Sources** panel UI) will not be persisted after an app reload — making some code paths difficult to pause on via a breakpoint.

### Workarounds

- Interactively set breakpoints do work, semi-reliably, via [Fast Refresh](http://localhost:3000/docs/fast-refresh).
- For now, inserting a `debugger;` statement in code is the most reliable way to pause on breakpoints.

**Comment by huntie** [comment]
## [Known issue][Frontend] Clicking the Chrome app icon in the macOS dock opens separate empty window

When navigating back to an opened debugger window on macOS using the dock icon, this will foreground the app window but also will open a new empty browser window in front.

<img width="1381" alt="image" src="https://github.com/react-native-community/discussions-and-proposals/assets/2547783/d650e438-3fcd-432d-9ed5-e9dcfe6c51d8">

This appears to be an issue/limitation with passing the [`--app` flag](https://peter.sh/experiments/chromium-command-line-switches/#app) to Chrome (via [chrome-launcher](https://github.com/GoogleChrome/chrome-launcher)). We're open to suggestions/contributions for fixing this!

**Reply by matteohcf** [reply]
same problem, any solution?

**Reply by huntie** [reply]
Neither us or Expo have solved this one yet 😅. We reckon the solution will be to package the DevTools frontend as its own standalone application (potentially, an Electron app) — having reverted a Chrome guest profile approach which nearly solved this internally. Unfortunately, solving this exactly is a lower priority than getting other debugger features ready and stable.

**Reply by huntie** [reply]
We've resolved this in React Native DevTools by reverting to the approach of asking Chrome to open an app window within the context of your primary Chrome profile.

- ✅ This means no separate dock icon, eliminating this issue.
- ℹ️ This trades off for other unfavourable behaviours: no easy `cmd+tab` on macOS, and the window size we request from Chrome is ignored when there is 1+ other window open.

We aim to evolve and improve this in future, but for now it was medium-to-low pri.

**Comment by efstathiosntonas** [comment]
Hey folks, tried to run the debugger but I get `error: unknown command 'start'`

![Screenshot 2023-12-07 at 20 18 43](https://github.com/react-native-community/discussions-and-proposals/assets/717975/0c82ba1c-1bbc-448a-8ae3-278f9ec7b19c)

EDIT: nvm, I was running the command outside of the project 😵‍💫

**Comment by boorad** [comment]
Am I doing something wrong in Chrome Console to suppress lines/multiline, or should there be more of the error showing?

<img width="1787" alt="Screenshot 2023-12-07 at 3 41 09 PM" src="https://github.com/react-native-community/discussions-and-proposals/assets/15431/4f1c6a8b-9989-41a9-a95a-af37c0fdbdab">

**Reply by buschco** [reply]
I can confirm on `0.74.0 rc-6` multiline console logs are not working properly

Code I tested:

```jsx
<Button
  onPress={() => {
    console.info(JSON.stringify({foo: {bar: {baz: [{foo: '123'}]}}}));
    console.log(JSON.stringify({foo: {bar: {baz: [{foo: '123'}]}}}));
    console.warn(JSON.stringify({foo: {bar: {baz: [{foo: '123'}]}}}));
    console.debug(JSON.stringify({foo: {bar: {baz: [{foo: '123'}]}}}));
    console.error(JSON.stringify({foo: {bar: {baz: [{foo: '123'}]}}}));
  }}
  title="Test Log"
/>
```

<img width="1209" alt="Screenshot 2024-04-11 at 12 01 58" src="https://github.com/react-native-community/discussions-and-proposals/assets/14929556/365829ce-fdda-4696-8f73-edbe47a985fc">

**Reply by huntie** [reply]
Thanks for highlighting this. We've confirmed this was a regression 0.73 and will be pursuing a fix.

- https://github.com/facebook/react-native/issues/43258
- https://github.com/facebook/react-native/issues/43838

**Comment by byCedric** [comment]
# Sources tab doesn't render loaded files

The React Native JS inspector uses a relatively modern version of Chrome DevTools. This version includes the revamped "File system" list, under the Sources tab. Unfortunately, this is not fully functional and doesn't show any useful information about all loaded files within the bundle.

![Current sources tab](https://github.com/react-native-community/discussions-and-proposals/assets/1203991/12a6f5a2-8c28-4960-8215-6e546b46edb1)

The _only_ way to open a bundled file is through `cmd+p`. This isn't something everyone is likely to find.

![cmd+p action](https://github.com/react-native-community/discussions-and-proposals/assets/1203991/42390454-bfd0-4467-8163-f3e8b3ba3eb4)

Ideally, the "Page" file tree would work for React Native apps. 

![Web's "Page" panel under the Sources tab](https://github.com/react-native-community/discussions-and-proposals/assets/1203991/5032f3f9-d228-4866-b636-532e11c00d37)

During some experimenting, I think we have to support a couple of different CDP domains to get this working: [`Page`](https://cdpstatus.reactnative.dev/devtools-protocol/tot/Page), [`Target`](https://cdpstatus.reactnative.dev/devtools-protocol/tot/Target), and [`IO`](https://cdpstatus.reactnative.dev/devtools-protocol/tot/IO).

I [hacked a basic implementation together](https://github.com/expo/expo/blob/%40bycedric/debugger/enable-pages/packages/%40expo/cli/src/start/server/metro/debugging/inspectorHandlers/PageFrame.ts) including the following events. Unfortunately, this still did NOT enable the desired "Page" panel under the Sources tab.
- Added `Page.frameStartedLoading`
- Added `Page.frameStoppedLoading`
- Added `Target.targetInfoChanged`
- Mutated `Runtime.executionContextCreated` (adding the `auxData` to this event)
- Mutated `Debugger.scriptParsed` (adding the `executionContextAuxData` to this event)

---

One very important side-effect of adding support for the events listed above was the [`Network.loadNetworkResource`](https://cdpstatus.reactnative.dev/devtools-protocol/tot/Network#method-loadNetworkResource) event being invoked from the React Native JS inspector. This seems to be the "proper" way of requesting external files from within the debugger but loading them from the device instead. This would actually solve the currently "dirty" workaround of inlining source maps as base64 encoded strings.

**Reply by evelant** [reply]
I'm also seeing this (or rather not seeing any sources in the sources tab). It seems the sources tab is completely broken/missing. Didn't realize that cmd+p would work. @huntie this has been broken since the introduction of the experimental debugger AFAIK and it caused me to think that the debugger was uselessly broken since I didn't realize cmd+p would work. Is this actively tracked by anyone working on the debugger?

**Comment by mmmoussa** [comment]
Under the memory tab, none of the snapshotting or sampling functionality works.

**Reply by kraenhansen** [reply]
FWIW, I was able to do a snapshot of the heap.

**Comment by mmmoussa** [comment]
I'm also unable to get profiling working. Only the console works as expected. I'm confused what functionality actually exists today aside from the debugger UI.

**Reply by evelant** [reply]
Same here, none of it seems to work beyond console and hitting explicit `debugger` statements in code. I think https://github.com/react-native-community/cli/pull/2316 was probably premature -- now there is absolutely no way to take performance profiles at all. profile-hermes was removed from the cli and profiling doesn't even seem to be enabled in the new debugger.

**Comment by handipriyono** [comment]
Thanks,  i just need the console.log feature right now.
👍🏽  <img width="769" alt="Screenshot 2023-12-25 at 3 32 17 PM" src="https://github.com/react-native-community/discussions-and-proposals/assets/35618196/df70aca3-85ac-4614-a18b-6f53b64370a0">

**Comment by totodot** [comment]
Any possibility to network debugging?

**Reply by huntie** [reply]
This is in our future aspirational plans. **For Expo users**, network debugging (via Expo's existing CDP implementation) will be available soon — see more [here](https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/2).

**Reply by Rossella-Mascia-Neosyn** [reply]
and for which does not use expo?

> This is in our future aspirational plans. **For Expo users**, network debugging (via Expo's existing CDP implementation) will be available soon — see more [here](https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/2).

**Reply by weltkind6** [reply]
waiting for network debugging!

**Comment by navjotsrmtpl** [comment]
in absence of network and application debugging it is becoming difficult to debug things. Any update when it would get resolved?

**Comment by meftunca** [comment]
You can use this temporarily [my repo](https://github.com/meftunca/reactotron-with-network-inspector)
[pr](https://github.com/infinitered/reactotron/pull/1426)

**Comment by maxphillipsdev** [comment]
Looks awesome! Love the idea of having the CDP server persist the session between bundle changes. Do you know if it will be possible to bundle the new tools into release builds?

**Comment by floydkim** [comment]
JS `Map` and `Set` objects are not printing to the console.

```ts
console.log('map', new Map([['a', 1], ['b', 2]]))
console.log('set', new Set([1, 2, 3, 4, 5]))
```

<img width="1289" alt="image" src="https://github.com/react-native-community/discussions-and-proposals/assets/22050211/9939ed62-442a-4d8f-b3db-c60eb1dc2cfd">

tested on `react-native-test-app` iOS app (react-native version 0.73.3, Hermes enabled)

**Reply by Wayne-Kim** [reply]
Hi~ floyd ㅋㅋㅋㅋ

**Reply by floydkim** [reply]
@WayneKim92 hi 😄 

discussion에 코멘트 달리면 참여자들 모두에게 github 알림이 가더라구요 😅  주의 요망.. ㅎㅎ

**Comment by huntie** [comment]
## [FAQ] Network debugging

This has been a popular question, so this is a dedicated comment on the status of this feature.

### TLDR: Available in Expo now, available for other apps in the future

The Network panel in Chrome DevTools is currently supported in Expo [since SDK 49](https://docs.expo.dev/debugging/tools/#inspecting-network-requests), but not in core React Native. We envision having Network support for all React Native apps, **but are sequencing this work later in our roadmap**.

### Motivation

As [mentioned above](https://github.com/react-native-community/discussions-and-proposals/discussions/733#discussion-5861632), the React Native team are currently focused on reimplementing React Native's debugger server (an in-place replacement of the Hermes debugger), giving us a modern debugging platform base that is fundamentally **more reliable and more extendable**.

> [We are making] fundamental improvements [...] to our backend debugging stack. Among other things, these future improvements aim to:
> 
> - **_Enable us to build new debugging capabilities on top of this stack in future._**

After this point, we will be in a position to build out our debugging features, with supporting the Network panel high up in our plans.
- We are not investing effort in migrating Expo's Network support to the current debugging experience in the meantime — since it will be partially/fully incompatible against our rearchitected server.

At this time, we aren't committing to a timeline for Network debugging in React Native core. We are collaborating with the Expo team on future improvements to the debugger (including new Network support), with React Native and Expo sharing a common implementation further down the line.

### Updates

- Sep 2024: The Network panel is **work in progress** (no ETA), and will come down the line to [React Native DevTools](https://github.com/react-native-community/discussions-and-proposals/discussions/819).

**Comment by pgpankajcapgemini** [comment]
Is this support for Physical device iPad or iPhone because for physical I am getting 

'''''
warn No compatible apps connected.
''''

**Reply by huntie** [reply]
Yes, we support physical devices ✅. So long as you are serving your app to the connected device via Metro, this should work. Please review the [Running on Device](https://reactnative.dev/docs/running-on-device?platform=ios) guide (including troubleshooting) — and if this still doesn't work, consider opening a new issue (you can debug by querying `http://localhost:8001/json/list` when the dev server is running).

**Comment by Asadaial** [comment]
how i can use Redux extentions in this debugger, just like Flipper?

**Reply by huntie** [reply]
Unfortunately, supporting third party Chrome extensions are not in our immediate plans. We're prioritising building a stable core debugging experience and a first party integration with React DevTools.

We understand that this leaves a tooling gap for popular developer extensions such as Redux — for the time being you will need to use an alternative debugger (e.g. [`react-native-debugger`](https://github.com/jhen0409/react-native-debugger)), or debug a React Native Web app.

**Reply by Lakston** [reply]
except that react native debugger can not be used anymore with v73+, so we lost network and redux debugging in the process.

I just updated our app tp v73 and I cannot debug the redux state or see network request, and the compatibility of RN with tools like react native debugger or reacttotron was removed, so it is a straight up downgrade.

**Reply by buschco** [reply]
>  we lost [...] redux debugging in the process

you can add redux devtools yourself. Have a look here https://github.com/reduxjs/redux-devtools

Basically you need to

1. install `@redux-devtools/cli` and `@redux-devtools/remote` 
2. enhance your compose:
```
const composeEnhanced = require('@redux-devtools/remote').composeWithDevTools({
  secure: false,
  realtime: true,
  hostname: 'localhost',
  port: 8000,
})
```

3. start the frontend `redux-devtools --hostname=localhost --port=8000 --open` connect somehow 😂

**Reply by omniphx** [reply]
@buschco, did you ever figure out how to connect? I'm using Redux Toolkit, which is supposed to connect to redux devtools automatically: https://redux-toolkit.js.org/api/configureStore#devtools

**Reply by buschco** [reply]
@omniphx

Just as a disclaimer, I never worked with redux tool kit.

From flying over the docs it seems, the connection is established by setting some global variables. However, since you can not do remote JS execution any more, you will have to find another way.

The way I described above is using a websocket connection, from the react-native app to the devtools remote. I think you need to find a way to force redux tool kit to do the same.

**Reply by fabiendem** [reply]
Hi, would you be able to provide some guidance if one wants to integrate a tool such as https://github.com/apollographql/apollo-client-devtools into the debugger please?
Original discussion at https://github.com/apollographql/apollo-client-devtools/issues/1420

**Reply by huntie** [reply]
@fabiendem For now, our debugger frontend will not support loading/embedding of Chrome DevTools plugins — although our long term vision is for this to be possible.

In the meantime, [Expo DevTools Plugins](https://docs.expo.dev/debugging/devtools-plugins/) may help solve your use case.

I'll link to [my other comment](https://github.com/react-native-community/discussions-and-proposals/discussions/734#discussioncomment-9904373) until I can share more here in future.

**Reply by fabiendem** [reply]
Sounds good, understood @huntie. Thank you!

**Comment by FouadMagdy01** [comment]
cannot copy objects from the console

**Reply by huntie** [reply]
I don't think there's a bug here, as we match behaviour with Chrome. Chrome's "Copy object" menu item captures a minimal plain text representation.

**Chrome 123 (web)**

<img width="653" alt="image" src="https://github.com/react-native-community/discussions-and-proposals/assets/2547783/053c7df4-71f3-4726-99be-0768974ea70c">

**React Native (Hermes/New Debugger)**

<img width="474" alt="image" src="https://github.com/react-native-community/discussions-and-proposals/assets/2547783/2869bf27-f9ac-4f02-89f8-0956005b02b1">

**Reply by FouadMagdy01** [reply]
Try to right click the object in the new debugger and then click ```copy object``` and paste it and you will just see ```{``` in the place you paste the object
Please give me a confirmation if this bug is not reproducible

**Reply by FouadMagdy01** [reply]
> cannot copy objects from the console

I can confirm that this issue is fixed in the latest release

**Comment by huntie** [comment]
## [Known issue] Keyboard shortcuts for zoom controls are nonfunctional 

https://github.com/facebook/react-native/issues/44595

**Comment by EduFrazao** [comment]
Its possible to map dynamically loaded scripts, evaluated by Function constructor?
Our app makes heavy use of scriptlets loaded from our backend to quick dispatch small changes, and will be GREAT to see some support for this, like Mozilla Firefox does:

https://firefox-source-docs.mozilla.org/devtools-user/debugger/how_to/debug_eval_sources/index.html

**Comment by Wayne-Kim** [comment]
If you need to debug network.

you can use it. [proxyman](https://proxyman.io/)

refer to this document to set up the tool: https://docs.proxyman.io/debug-devices/react-native
pin your domain and app in proxyman

<img width="279" alt="스크린샷 2024-07-05 오후 3 06 38" src="https://github.com/react-native-community/discussions-and-proposals/assets/75321423/54137003-9642-4f29-88c4-2226b80a1d89">

**Comment by Craques** [comment]
Managed to view network logs using requestly. Both for iOS and Android

https://developers.requestly.com/guides/debugging-https-traffic-in-ios-simulator/

Also tested using android studio network inspector and it seems to work 

https://developer.android.com/studio/debug/network-profiler

**Reply by amerllica** [reply]
The [Charles](https://www.charlesproxy.com/) does it the same.

**Comment by amerllica** [comment]
I'm using [`react-native-network-logger`](https://www.npmjs.com/package/react-native-network-logger) temporarily and wait for the `network` tab enthusiastically.

Really it's a tough moment in which we haven't got any official tool to debug network interactions. The `react-native-network-logger
` does provide a screen within the app to see the network's stuff but imagine, for debugging, a debug page was added to the project.

Peculiar, but real!!!

**Comment by anis-18** [comment]
Can we make the ui like flipper ? It was very confortable and user expérience was great.
I find a little challenging with current ui

**Comment by AndreiCalazans** [comment]
When creating an app with the CLI and trying to open the experimental debugger the app opens a Chrome window but nothing loads and the Metro running freezes. 

app verions:
"react": "18.3.1",
"react-native": "0.75.2"

Screenshots:

<img width="797" alt="image" src="https://github.com/user-attachments/assets/b14c5724-90a5-44f8-9075-eba851161ddd">


<img width="700" alt="image" src="https://github.com/user-attachments/assets/18de8370-e9f6-4e91-b113-0df74309e998">

**Comment by rawatnaresh** [comment]
Hi, Nothing happened when I set Network throttling to offline.

<img width="810" alt="Screenshot 2024-09-16 at 14 44 25" src="https://github.com/user-attachments/assets/0dcf7db4-709a-4eeb-8f5a-19c3578f113e">

**Reply by huntie** [reply]
The Network panel isn't provided by our frontend or supported by React Native at this time — did you open this in Chrome? Going forward, direct Hermes debugging via Chrome isn't one of our supported flows. **When** we do add Network debugging support to React Native, it will become available in our frontend.

**Reply by rawatnaresh** [reply]
Yes, I opened it in chrome.

**Comment by alexfov** [comment]
Is it possible to add an own plugin?

**Reply by huntie** [reply]
Not for now. We're exploring the ability to load third party Chrome extensions, but this won't be coming any time soon.

**Comment by antoniovuono** [comment]
Is possible install the debugger in oldest versions of react-native ?

**Reply by huntie** [reply]
Hi, unfortunately not. (Sidenote: Thanks, I'll note this as an FAQ!)

**Comment by fabiendem** [comment]
It would help to see a table of the features available in the new debugger.
Below is my understanding, but I am not sure it's accurate ⚠️ 

Debugging features supported by the new debugger:

Feature|React Native|Expo|Alternatives
---|---|---|---
Console logs | ✅  | ✅ | [React Native commands](https://reactnative.dev/docs/debugging-native-code#accessing-logs)
Network profiling | 🛠️  | ✅ | [Reactotron](https://docs.infinite.red/reactotron/plugins/networking/)<br/>[Proxyman](https://proxyman.io/)<br/>[iOS Instruments](https://www.kodeco.com/27390649-monitoring-http-traffic-with-instruments)<br/>[Android Studio](https://developer.android.com/studio/debug/network-profiler)
Breakpoints | ✅  | ✅ | [Hermes Debugger](https://reactnative.dev/docs/debugging?js-debugger=hermes#opening-the-debugger)
Profiler | ✅  | ✅ | [React Devtools](https://reactnative.dev/docs/debugging?js-debugger=new-debugger#react-devtools)
Plugins | 🛠️   | 🛠️ | 
What else? | |

Feel free to take the table over and improve/correct it 🙏

**Reply by EduFrazao** [reply]
Guys, its possible to take a look on this feature?

https://github.com/react-native-community/discussions-and-proposals/discussions/733#discussioncomment-9938143

It would be very nice to be possible to debug dynamic loaded snippets.

**Comment by cawfree** [comment]
Thanks for this amazing work.

I'm using [`react-native-threads`](https://github.com/joltup/react-native-threads) pulls JavaScript from Metro and then executes via a dedicated Catalyst instance.

I've found that by fetching the JavaScript bundle using [`loadSplitBundleFromServer`](https://github.com/facebook/react-native/blob/23f62acd839d38d7ea4ba9d7d2e99bb2a1aa0bb9/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/BridgeDevSupportManager.java#L93C15-L93C40), I can get as far as seeing my asynchronously loaded `self.thread.js` (and it's corresponding `self.thread.bundle`) in the debugger window, however it does not appear to detect any console output (nor do the breakpoints work).

<img width="352" alt="image" src="https://github.com/user-attachments/assets/5a1ab4dd-46f1-41c0-8dd6-c6509196433d">

Likewise, the contents of the debugger window only seem to populate if I load from Metro as `modulesOnly=false`, but I need to treat the bundle as an executable instance. I've tried hacking together dedicated `BridgeDevSupportManager`s with unique keys to see if that would fool the bundler into opening a separate context for the bundle, but no dice.

Do any minimal examples exist on how to setup an isolated debugging session for an arbitrary file bundle?

**Comment by riyal-abhay** [comment]
Issue with Experimental Debugger on Android Emulators in React Native 0.73

Hey Guys,
We recently upgraded our project to React Native 0.73 and were really excited to try out the experimental Debugger. It works flawlessly with iOS simulators, and we appreciate the improvements made.
However, we are encountering an issue when attempting to use the experimental Debugger on Android emulators. The Debugger fails to initialize, and we are met with the following error message:

<img width="617" alt="Screenshot 2024-09-30 at 7 10 14 AM" src="https://github.com/user-attachments/assets/6b177874-e228-456c-bda0-4a65918e7cb3">

**Comment by OmkarK45** [comment]
### Introduction

Hi, 
The new debugger is great! However, it'd be great if it allowed non-guest logging in option. I use chrome as my daily driver and have few extensions and themes within chrome that make my life easier. In its current state, it is not possible to install extensions within the new experimental debugger. 

### Details
- An option/flag to allow logging into chrome as a proper user. 
- It'll allow installation of useful extensions in new experimental debugger 
- It'll allow extensions to load custom stylesheets that significantly improves the font size, interface of the dev tools itself. 

### Screenshots
<img width="1210" alt="image" src="https://github.com/user-attachments/assets/9ca2246e-0bbd-488d-bdf7-284988f29ba4">

**Reply by OmkarK45** [reply]
PS: I've gone through the above thread and it looks like these features are low priority right now — no worries

**Comment by Ajay-Quant** [comment]
This is my package.json and the issue that I am facing is that when I am trying to start the devtools, my app suddenly crashes without any logs

this is my package.json

`{
  "name": "Airline",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "@react-navigation/bottom-tabs": "^7.2.0",
    "@react-navigation/elements": "^2.2.5",
    "@react-navigation/native": "^7.0.14",
    "@react-navigation/native-stack": "^7.2.0",
    "@types/lodash": "^4.17.14",
    "@types/react-native-global-props": "^1.1.6",
    "axios": "^1.7.9",
    "clsx": "^2.1.1",
    "dayjs": "^1.11.13",
    "lodash": "^4.17.21",
    "lodash-es": "^4.17.21",
    "nativewind": "^4.1.23",
    "react": "18.3.1",
    "react-native": "0.76.5",
    "react-native-global-props": "^1.1.5",
    "react-native-linear-gradient": "^2.8.3",
    "react-native-reanimated": "^3.16.6",
    "react-native-responsive-screen": "^1.4.2",
    "react-native-safe-area-context": "^5.0.0",
    "react-native-screens": "^4.4.0",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "^15.10.1",
    "tailwindcss": "^3.4.17",
    "zustand": "^5.0.2",
    "@react-native-voice/voice": "3.2.4",
    "react-native-tts": "4.1.1"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/preset-env": "^7.25.3",
    "@babel/runtime": "^7.25.0",
    "@react-native-community/cli": "15.0.1",
    "@react-native-community/cli-platform-android": "15.0.1",
    "@react-native-community/cli-platform-ios": "15.0.1",
    "@react-native-community/netinfo": "11.4.1",
    "@react-native/babel-preset": "0.76.5",
    "@react-native/eslint-config": "0.76.5",
    "@react-native/metro-config": "0.76.5",
    "@react-native/typescript-config": "0.76.5",
    "@types/react": "^18.2.6",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.6.3",
    "eslint": "^8.19.0",
    "jest": "^29.6.3",
    "prettier": "2.8.8",
    "react-test-renderer": "18.3.1",
    "typescript": "5.0.4"
  },
  "engines": {
    "node": ">=18"
  }
}
`

any help please
